ocean-client/src/vault-maxi.ts

```diff
@@ -38,82 +41,47 @@ export async function main(event: maxiEvent): Promise<Object> {
-                console.log(message)
-                await telegram.log(message)
-                await telegram.send(message)
-                return {
-                    statusCode: 500,
-                    message: message
-                }
-            }
+            let result= await program.doAndReportCheck(telegram)
+            return { statusCode: result? 200 : 500 }
         }
     }
 
-    const program = new VaultMaxiProgram(store, new WalletSetup(MainNet, settings))
-    await program.init()
-    if (! await program.isValid()) {
-        await telegram.send("Configuration error. please check your values")
-        console.error("Configuration error. please check your values")
+    if (!await program.doValidationChecks(telegram)) {
         return {
             statusCode: 500
         }
     }
 
-    const vaultcheck = await program.getVault()
-    if(!vaultcheck) {
-        console.error("Did not find vault")
-        await telegram.send("Error: vault is gone ")
-        return {
-            statusCode: 500
-        }
-    }
-    if (vaultcheck.state == LoanVaultState.FROZEN || vaultcheck.state == LoanVaultState.IN_LIQUIDATION) {
-        await telegram.send("Error: vault not active, its " + vaultcheck.state)
-        console.error("Vault not active: "+vaultcheck.state)
-        return {
-            statusCode: 500
-        }
-    }
 
-    let vault: LoanVaultActive = vaultcheck
-    if(+vault.collateralValue < 10) {
-        await telegram.send("less than 10 dollar in the vault, can't work with that")
-        console.error("less than 10 dollar in the vault. can't work like that")
-        return {statusCode:500}
-    }
     let result = true
+    let vault: LoanVaultActive = await program.getVault() as LoanVaultActive //already checked before if all is fine
+    
 
     // 2022-03-08 Krysh: Something went wrong on last execution, we need to clean up, whatever was done
-    if (settings.stateInformation && settings.stateInformation.state !== ProgramState.Idle) {
+    if (settings.stateInformation.state !== ProgramState.Idle) {
         const information = settings.stateInformation
         console.log("last execution stopped state " + information.state)
         console.log(" for tx " + information.tx)
         console.log(" for txId " + information.txId)
         console.log(" on block height " + information.blockHeight)
 
-        let shouldCleanUp = false
-
         // 2022-03-09 Krysh: input of kuegi
         // if we are on state waiting for last transaction and we are still on same block height since last execution
         // then we should wait for txId
-        const currentBlockHeight = await program.getBlockHeight()
-        if (information.state === ProgramState.WaitingForTransaction && information.blockHeight === currentBlockHeight) {
+        if (information.state === ProgramState.WaitingForTransaction) { //ignore blockheight, just ensure that the tx is done
```

Comment:
good point that we won't need block height anymore, if we have txId directly. Maybe we should remove it completely? Also please adjust the comment above. Thanks